src/gui/userstatus.cpp

```diff
@@ -28,43 +29,53 @@ namespace OCC {
 
 Q_LOGGING_CATEGORY(lcUserStatus, "nextcloud.gui.userstatus", QtInfoMsg)
 
-UserStatus::UserStatus(QObject *parent)
-    : QObject(parent)
-{
-}
-
-UserStatus::Status UserStatus::stringToEnum(const QString &status) const 
-{
-    // it needs to match the Status enum
-    const QHash<QString, Status> preDefinedStatus{{"online", Status::Online},
-                                               {"dnd", Status::DoNotDisturb}, //DoNotDisturb
-                                               {"away", Status::Away},
-                                               {"offline", Status::Offline},
-                                               {"invisible", Status::Invisible}};
+namespace {
+    UserStatus::Status stringToEnum(const QString &status) 
+    {
+        // it needs to match the Status enum
+        const QHash<QString, UserStatus::Status> preDefinedStatus{
+            {"online", UserStatus::Status::Online},
+            {"dnd", UserStatus::Status::DoNotDisturb},
+            {"away", UserStatus::Status::Away},
+            {"offline", UserStatus::Status::Offline},
+            {"invisible", UserStatus::Status::Invisible}
+        };
+        
+        // api should return invisible, dnd,... toLower() it is to make sure 
+        // it matches _preDefinedStatus, otherwise the default is online (0)
+        return preDefinedStatus.value(status.toLower(), UserStatus::Status::Online);
+    }
     
-    // api should return invisible, dnd,... toLower() it is to make sure 
-    // it matches _preDefinedStatus, otherwise the default is online (0)
-    const auto statusKey = status.isEmpty() ? QStringLiteral("online") : status.toLower();
-    return preDefinedStatus.value(statusKey, Status::Online);
+    QString enumToString(UserStatus::Status status) 
+    {
+        switch (status) {
+        case UserStatus::Status::Away:
+            return QObject::tr("Away");
+        case UserStatus::Status::DoNotDisturb:
+            return QObject::tr("Do not disturb");
+        case UserStatus::Status::Invisible:
+        case UserStatus::Status::Offline:
+            return QObject::tr("Offline");
+        case UserStatus::Status::Online:
+            return QObject::tr("Online");
+        default:
```

Comment:
remove `default` completely and put the Q_UNREACHABLE call after the end of the switch
that way we can get warnings at compile time if the enum gets more values